sw/device/lib/dif/dif_kmac.h

```diff
@@ -375,14 +382,74 @@ typedef enum dif_kmac_error {
  * The state of the message FIFO used to buffer absorbed data.
  */
 typedef enum dif_kmac_fifo_state {
-  /** The message FIFO is empty. */
-  kDifKmacFifoStateEmpty,
   /** The message FIFO is not empty or full. */
-  kDifKmacFifoStatePartial,
+  kDifKmacFifoStatePartial = 0,
+  /** The message FIFO is empty. */
+  kDifKmacFifoStateEmpty = 1 << 0,
   /** The message FIFO is full. Further writes will block. */
-  kDifKmacFifoStateFull,
+  kDifKmacFifoStateFull = 1 << 1,
 } dif_kmac_fifo_state_t;
 
+typedef enum dif_kmac_sha3_state {
+  /**
+   * SHA3 hashing engine is in idle state.
+   */
+  kDifKmacSha3StateIdle = 1 << 0,
+
+  /**
+   * SHA3 is receiving message stream and processing it.
+   */
+  kDifKmacSha3StateAbsorbing = 1 << 1,
+
+  /**
+   * SHA3 completes sponge absorbing stage. In this stage, SW can manually run
+   * the hashing engine.
+   */
+  kDifKmacSha3StateSqueezing = 1 << 2,
+} dif_kmac_sha3_state_t;
+
+typedef enum dif_kmac_alert_faults {
+  /**
+   * Neither errors nor fault has occurred.
```

Comment:
Would it make more sense to have a struct of bools?

Reply:
I think that in this case both solutions are equivalent.
So I'll leave this way. Thanks